jetscape_analysis/analysis/analyze_events_base_STAT.py

```diff
@@ -72,21 +72,24 @@ def __init__(self, config_file="", input_file="", output_dir="", **kwargs):
             _final_state_hadrons_path = Path(self.input_file_hadrons)
             # For an example filename of "jetscape_PbPb_Run0005_5020_0001_final_state_hadrons_00.parquet",
             # - the run number is index 2
-            _job_identifier = _final_state_hadrons_path.stem.split("_")[2]
-            if "Run" in _job_identifier:
-                # We're using a standard production with a run number - look for the run info file.
-                _run_number = _job_identifier
-                # - the file index is at index 4 (in the example, it extracts `1` as an int)
-                _file_index = int(_final_state_hadrons_path.name.split('_')[4])
-                run_info_path = _final_state_hadrons_path.parent / f"{_run_number}_info.yaml"
-                with open(run_info_path, 'r') as f:
-                    _run_info = yaml.safe_load(f)
+            _run_number = _final_state_hadrons_path.stem.split("_")[2]
+            # - the file index is at index 4 (in the example, it extracts `1` as an int)
+            _file_index = int(_final_state_hadrons_path.name.split('_')[4])
+            run_info_path = _final_state_hadrons_path.parent / f"{_run_number}_info.yaml"
+            with open(run_info_path, 'r') as f:
+                _run_info = yaml.safe_load(f)
+
+                self.centrality_range = _run_info["centrality"]
+                # Default to 'precomputed_hydro' if not specified to ensure backward compatibility
+                self.soft_sector_execution_type = _run_info.get("soft_sector_execution_type", "precomputed_hydro")
+
+                if self.soft_sector_execution_type == "precomputed_hydro":
                     centrality_string = _run_info["index_to_hydro_event"][_file_index].split('/')[0].split('_')
                     # index of 1 and 2 based on an example entry of "cent_00_01"
                     self.centrality = [int(centrality_string[1]), int(centrality_string[2])]
-            else:
-                # No run info available - need to retrieve the centrality event-by-event
-                self.use_event_based_centrality = True
```

Comment:
This is to support no run_info being available, so I think this needs to be restored